### DISCUSSION TITLE
Homeassistant MQTT autodiscover (missing entities)

### DISCUSSION THREAD

**Discussion by mmick08** [goecharger/go-eCharger-API-v2#234] [General]
Hi Guys,

First of all, thank you for integrating Homeassistant MQTT directly in your firmware. A great step underlining your dedication to open source.

I just wanted to point out the following topics:

**Missing entities:**

Even if the auto discover works very well several important entities to do home automation are missing.
i.e. "Car Connected" (is the cable plugged in or not)
      "Configured phases" (how many phases are currently used)
      "phase switching" (possibility to choose between automatic, 1-phase, 3-phases)

**Wrong entity details:**

As for now the "total_energy_charged" cannot be used within the Energy dashboard as state_class: total_increasing is missing for the entity.

Thank you for taking into account :)

BR Manuel

**Comment by FransOv** [comment]
Yesterday I switched to your MQTT integration, I formerly used the https://github.com/syssi/homeassistant-goecharger-mqtt integration. I noticed that the integration was discovered by HA even before I enabled the integration in the app. Works quit well, but the above mentioned lack of state_class total_increasing for total_energy_charged was a problem. I solved it for now by creating a helper template entity based on total_energy_charged with the required device_class and state_class. I successfully moved the statistics of the old sensor of the sissy integration to the new template sensor by changing the metadata_id in the statistics and the statistics_short_term tables in the HA database.
